### ISSUE TITLE
todos

### ISSUE THREAD

**Issue by aobo-y** [aobo-y/SouthPark-Chatbot#1]
- [ ] implement pre-trained word embedding @aobo-y 
- [x] south park/the simpsons data preprocessing @jiahao42 
- [x] implement Beam search in construct response @jiahao42 
- [x] pretrain with movie data, then train with southpark data @ddddwy 
- [x] personal embedding @ddddwy 
- [ ] implement Blue @quq99
- [ ] try Ocean @quq99 

lower priority
- [ ] train glove embedding with our dataset @aobo-y 
- [ ] support LSTM
- [ ] auto load latest possible checkpoint
- [ ] return responses generated by beam search randomly @jiahao42 
- [ ] handle `'m` `'ve` `'re`.... in normalize

**Comment by ddddwy** [comment]
Suggested new to do list:

- [ ] southpark data preprocessing
- [ ] insert personal embedding, pretrained word embedding
- [ ] add evaluation metrics: BLUE, perplexity, OCEAN(maybe)

**Comment by ddddwy** [comment]
> I tried 2 pretrained [GloVe embedding ](https://github.com/stanfordnlp/GloVe#download-pre-trained-word-vectors) and filtered it with our corpus, the result is as below
> 
> ### glove.42B.300d.zip
> size of the data: 49928
> size of the embedding: 1917495
> size of the filtered embedding: 45116
> 
> ### glove.6B.zip
> size of the data: 49928
> size of the embedding: 400001
> size of the filtered embedding: 40724
